### DISCUSSION TITLE
readonly upper branch and read/write lower branch

### DISCUSSION THREAD

**Discussion by IdrisMiles** [trapexit/mergerfs#1101] [Q&A] [answered]
Hello, I'm trying to understand if MergerFS is suitable for my use case, my initial testing has proved unsuccessful but I may not be setting all the appropriate configuration to achieve my goal - or MergerFS is not intended for my use case.

The use case:
I have 2 identical filesystems acting in a master/replica setup, where one filesystem is read-only but fast to read from, the other is read-write but slow to read from.
My goal is to merge these 2 filesystems so that all reads are directed to the fast RO filesystem and writes are directed to the slow RW filesystem.

I've attempted the following:
```bash
mkdir readonly readwrite merged
echo "hello" > readonly/file.txt
echo "hello" > readwrite/file.txt

mergerfs -o allow_other,use_ino /tmp/readonly=RO:/tmp/readwrite=RW /tmp/merged

echo "world" >> merged/file.txt

cat merged/file.txt
# hello
# world

cat readonly/file.txt
# hello
# world

cat readwrite/file.txt
# hello
```

Firstly, this seems odd to me as the `=RO` filter does not seem to be respected?
Secondly, my desired results would actually be:
```bash
cat merged/file.txt
# hello

cat readonly/file.txt
# hello

cat readwrite/file.txt
# hello
# world
```

It may seem counter-intuitive to want the merged path to show the old contents but another process is in charge of syncing the changes to the readonly filesystem, so they would eventually become available.

Is what I'm hoping to achieve possible with MergerFS?

I've had a brief look at the policies and tried some variations but to no avail, all of the below produce the same results:
```bash
mergerfs -o allow_other,use_ino,category.search=ff,category.create=ff,category.action=ff /tmp/readonly=RO:/tmp/readwrite=RW /tmp/merged

mergerfs -o allow_other,use_ino,category.search=ff,category.create=all,category.action=all /tmp/readonly=RO:/tmp/readwrite=RW /tmp/merged

mergerfs -o allow_other,use_ino,category.search=ff,category.create=newest,category.action=newest /tmp/readonly=RO:/tmp/readwrite=RW /tmp/merged

mergerfs -o allow_other,use_ino,category.search=ff,category.create=epff,category.action=epff /tmp/readonly=RO:/tmp/readwrite=RW /tmp/merged
```

Some system details:
mergerfs version: 2.33.5
fusermount version: 2.9.2
CentOS Linux release 7.9.2009 (Core)

**Comment by trapexit** [comment, marked as answer]
https://github.com/trapexit/mergerfs#how-it-works

> mergerfs does NOT support the copy-on-write (CoW) or whiteout behaviors found in aufs and overlayfs. You can not mount a read-only filesystem and write to it.

**Reply by trapexit** [reply]
https://github.com/trapexit/mergerfs#does-mergerfs-support-cow--copy-on-write--writes-to-read-only-filesystems

**Reply by IdrisMiles** [reply]
Ok so it's not suitable - thank you.

**Reply by trapexit** [reply]
No. You should use overlayfs.
